hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/token/impl/entity/AccountBuilderImpl.java

```diff
@@ -0,0 +1,248 @@
+import com.hedera.node.app.service.token.entity.AccountBuilder;
+import com.hedera.node.app.spi.key.HederaKey;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * An implementation of {@link AccountBuilder} for building Account instances. This class is
+ * <strong>not</strong> exported from the module.
+ */
+public class AccountBuilderImpl implements AccountBuilder {
+    private final Account copyOf;
+
+    // These fields are the ones that can be set in the builder
+    @SuppressWarnings("OptionalUsedAsFieldOrParameterType")
+    private Optional<HederaKey> key;
+
+    private long expiry;
+    private long balance;
+    private String memo;
+    private boolean deleted;
+    private boolean receiverSigRequired;
+    private long numberOfOwnedNfts;
+    private int maxAutoAssociations;
+    private int usedAutoAssociations;
+    private int numAssociations;
+    private int numPositiveBalances;
+    private long ethereumNonce;
+    private long stakedToMe;
+    private long stakePeriodStart;
+    private long stakedNum;
+    private boolean declineReward;
+    private long stakeAtStartOfLastRewardedPeriod;
+    private long autoRenewAccountNumber;
+    private long autoRenewSecs;
+
+    /**
+     * Create a builder for creating {@link Account}s, using the given copy as the basis for all
+     * settings that are not overridden.
+     *
+     * @param copyOf The instance to copy
+     */
+    public AccountBuilderImpl(@NonNull Account copyOf) {
+        this.copyOf = Objects.requireNonNull(copyOf);
+        this.key = copyOf.key();
+        this.expiry = copyOf.expiry();
+        this.balance = copyOf.balanceInTinyBar();
+        this.memo = copyOf.memo();
+        this.deleted = copyOf.isDeleted();
+        this.receiverSigRequired = copyOf.isReceiverSigRequired();
+        this.numberOfOwnedNfts = copyOf.numberOfOwnedNfts();
+        this.maxAutoAssociations = copyOf.maxAutoAssociations();
+        this.usedAutoAssociations = copyOf.usedAutoAssociations();
+        this.numAssociations = copyOf.numAssociations();
+        this.numPositiveBalances = copyOf.numPositiveBalances();
+        this.ethereumNonce = copyOf.ethereumNonce();
+        this.stakedToMe = copyOf.stakedToMe();
+        this.stakePeriodStart = copyOf.stakePeriodStart();
+        this.stakedNum = copyOf.stakedNum();
+        this.declineReward = copyOf.declineReward();
+        this.stakeAtStartOfLastRewardedPeriod = copyOf.stakeAtStartOfLastRewardedPeriod();
+        this.autoRenewAccountNumber = copyOf.autoRenewAccountNumber();
+        this.autoRenewSecs = copyOf.autoRenewSecs();
+    }
+
+    @Override
+    @NonNull
+    public AccountBuilder key(@NonNull HederaKey key) {
+        this.key = Optional.of(Objects.requireNonNull(key));
+        return this;
+    }
+
+    @Override
+    @NonNull
+    public AccountBuilder balance(long value) {
+        if (value < 0 || value > 50_000_000_000L) {
```

Comment:
Good idea! But `value` is in tinybars... 😛

Reply:
Great catch ! Thank you ! 😛